### PR TITLE
Use incremental updates more

### PR DESCRIPTION
**Pull request by noobpwnftw**
Moved

### CODE REVIEW

**Comment by noobpwnftw** [comment]
Fixes https://github.com/official-stockfish/Stockfish/pull/3149 too.
I have queued this version for testing just to confirm correctness.

File: src/nnue/nnue_feature_transformer.h [hunk 1]
```diff
@@ -289,11 +294,16 @@ namespace Eval::NNUE {
     // Calculate cumulative value using difference calculation
     void UpdateAccumulator(const Position& pos) const {
 
-      const auto prev_accumulator = pos.state()->previous->accumulator;
+      Accumulator* prev_accumulator;
+      if (pos.state()->previous->accumulator.computed_accumulation)
+        prev_accumulator = &pos.state()->previous->accumulator;
+      else
+        prev_accumulator = &pos.state()->previous->previous->accumulator;
```
Comment (vondele) [inline src/nnue/nnue_feature_transformer.h:301]: is this `pos.state()->previous->previous` always associated (also near the root?). Maybe good to put an assert there?

Comment (noobpwnftw) [reply src/nnue/nnue_feature_transformer.h:301]: Yes, to reach here without `pos.state()->previous->accumulator.computed_accumulation` being true, `pos.state()->previous->previous` must exist. I'll add assert.

**Comment by Sopel97** [comment]
I've seen the discusson on discord about making this cleaner. Instead of making a call in each branch which the compiler seems to have hard time to optimize I would try having an 2 element array for dps (using `ValueList<DirtyPiece, 2>` which is filled in the branches and the call is made only once after.

**Comment by noobpwnftw** [comment]
Unfortunately the cleanup version may not get optimized well, so I created this branch https://github.com/noobpwnftw/Stockfish/tree/incr_more7.

**Pull request closed without merging**